cmd/juju/storage/import.go

```diff
@@ -0,0 +1,154 @@
+that is in use by another Juju model.
+
+To import a filesystem, you must specify three things:
+
+ - the storage pool, which identifies the storage provider
+   which manages the storage; and with which the storage
+   will be associated
+ - the storage provider ID for the filesystem, or
+   volume that backs the filesystem
+ - the storage name to assign to the filesystem,
+   corresponding to the storage name used by a charm
+
+Once a filesystem is imported, Juju will create an associated storage
+instance using the given storage name.
+
+Examples:
+    # Import an existing filesystem backed by an EBS volume,
+    # and assign it the "pgdata" storage name. Juju will
+    # associate a storage instance ID like "pgdata/0" with
+    # the volume and filesystem contained within.
+    juju import-filesystem ebs vol-123456 pgdata
+`
+	importFilesystemCommandAgs = `
+<storage-provider|pool> <filesystem|volume-id> <storage-name>
+`
+)
+
+// importFilesystemCommand imports filesystems into the model.
+type importFilesystemCommand struct {
+	StorageCommandBase
+	newAPIFunc NewStorageImporterFunc
+
+	storagePool       string
+	storageProviderId string
+	storageName       string
+}
+
+// Init implements Command.Init.
+func (c *importFilesystemCommand) Init(args []string) error {
+	if len(args) < 3 {
+		return errors.New("import-filesystem requires a storage pool, provider ID, and storage name")
+	}
+	c.storagePool = args[0]
+	c.storageProviderId = args[1]
+	c.storageName = args[2]
+
+	if !storage.IsValidPoolName(c.storagePool) {
+		return errors.NotValidf("pool name %q", c.storagePool)
+	}
+
+	validStorageName, err := regexp.MatchString(names.StorageNameSnippet, c.storageName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if !validStorageName {
+		return errors.Errorf("%q is not a valid storage name", c.storageName)
+	}
+	return nil
+}
+
+// Info implements Command.Info.
+func (c *importFilesystemCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "import-filesystem",
+		Purpose: "Imports a filesystem into the model.",
+		Doc:     importFilesystemCommandDoc,
+		Args:    importFilesystemCommandAgs,
+	}
+}
+
+// Run implements Command.Run.
+func (c *importFilesystemCommand) Run(ctx *cmd.Context) (err error) {
+	api, err := c.newAPIFunc(&c.StorageCommandBase)
+	if err != nil {
+		return err
+	}
+	defer api.Close()
+
+	ctx.Infof(
+		"importing %q from storage pool %q as storage %q",
+		c.storageProviderId, c.storagePool, c.storageName,
+	)
+	storageTag, err := api.ImportStorage(
+		storage.StorageKindFilesystem,
+		c.storagePool, c.storageProviderId, c.storageName,
+	)
+	if err != nil {
+		return err
+	}
+	ctx.Infof("imported storage %s", storageTag.Id())
+	return nil
+}
+
+// StorageImporter provides a method for importing storage into the model.
+type StorageImporter interface {
+	Close() error
+
+	ImportStorage(
+		kind storage.StorageKind,
+		storagePool, storageProviderId, storageName string,
```

Comment:
I understand it's a bit stuttery, but I think it's important to include here because it's an interface. Removing "Storage" from the method name makes it just an "Importer", and relies only on the parameter types to distinguish from any other type of Import method. That makes the naming less obvious, to me at least.